access-grant/src/main/java/com/inrupt/client/accessgrant/Utils.java

```diff
@@ -119,6 +133,41 @@ public static List<Map<String, Object>> getCredentialsFromPresentation(final Map
         return credentials;
     }
 
+    public static int convertPage(final String page) {
+        try {
+            final int p = Integer.parseInt(page);
+            return Math.max(1, p);
+        } catch (final IllegalArgumentException ex) {
+            // ignore invalid values
+            LOGGER.debug("Invalid page value, defaulting to first page: {}", ex.getMessage());
```

Comment:
Could this cause an client to get into a continues loop if there was an error and 1 is returned?

Reply:
I'm not sure how this would result in a loop, unless a developer does some extremely odd things with the code they write (i.e. build a while loop, manipulate the next page filter with an random value and then pass that filter into the next `::query` call. But the developer would need to specifically manipulate the data within that loop. An alternative would be to throw an exception.

Reply:
Throwing an exception seems safer here.

Reply:
This code has been removed now that the fallback mechanism has been removed.